### ISSUE TITLE
`TypeVar` bounds do not work with `__class__`

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ty#662]
### Summary

This code does not type check [playground](https://play.ty.dev/143c2d06-491d-486f-abdf-77776d2f0215)
```py
from typing import TypeVar

_B = TypeVar("_B", bound="Bound")
class Bound:
    def copy(self: _B) -> _B:
        new = self.__class__()
        reveal_type(new)
        return new
```
`ty` gives
```
Revealed type: `Bound` (revealed-type) [Ln 7, Col 21]
Return type does not match returned value: expected `_B`, found `Bound` (invalid-return-type) [Ln 8, Col 16]
```
While `basedpyright` gives no errors and `Type of "new" is "_B@copy"` [playground](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgGoCGIAsAFC0D6AQlALyEkUgAUAREzwBooAIzABXFABMWPRuKk8AlLQDGAG3IBnTVDkTJALlpQTUScWBQVYBHC6bia4AahNFUALQA%2BV4yM1TQKgUYgB3VigHJwA6enp1LU04rmUAoJMQYgA3YnI1engEYi4Q0NT0jOIYMRAUYLCgA)
And `mypy` gives this [playground](https://mypy-play.net/?mypy=latest&python=3.13&gist=9f6fd8bbadc87087b1b44ad9b2cdf988)
```
main.py:7: note: Revealed type is "_B`-1"
Success: no issues found in 1 source file
```

### Version

ty 0.0.1-alpha.10 (15bae14c2 2025-06-13) + playground

**Comment by carljm** [comment]
This requires that we support `type[]` of a TypeVar, I think.
